src/client.ts

```diff
@@ -105,30 +110,91 @@ export class SkynetClient {
    * The Skynet Client which can be used to access Skynet.
    *
    * @class
-   * @param [portalUrl] The portal URL to use to access Skynet, if specified. To use the default portal while passing custom options, use ""
+   * @param [initialPortalUrl] The initial portal URL to use to access Skynet, if specified. A request will be made to this URL to get the actual portal URL. To use the default portal while passing custom options, pass "".
    * @param [customOptions] Configuration for the client.
    */
-  constructor(portalUrl: string = defaultPortalUrl(), customOptions: CustomClientOptions = {}) {
-    this.portalUrl = portalUrl;
+  constructor(initialPortalUrl = "", customOptions: CustomClientOptions = {}) {
+    if (initialPortalUrl === "") {
+      // Portal was not given, use the default portal URL. We'll still make a request for the resolved portal URL.
+      initialPortalUrl = defaultPortalUrl();
+    } else {
+      // Portal was given, don't make the request for the resolved portal URL.
+      this.givenPortalUrl = initialPortalUrl;
+    }
+    this.initialPortalUrl = initialPortalUrl;
     this.customOptions = customOptions;
   }
 
+  /**
+   * Make the request for the API portal URL.
+   *
+   * @returns - A promise that resolves when the request is complete.
+   */
+  /* istanbul ignore next */
+  async initPortalUrl(): Promise<void> {
```

Comment:
can be skipped if `customPortalUrl` is set right? only remark on this because in the comment you mentioned
`"don't make the request for the resolved portal URL."`